File: projects/Reserve/Reserve.Data/Elements/NavigatorElement.cs
using System.Collections.Generic;
using JetBrains.Annotations;
using Reserve.Data.Types;

namespace Reserve.Data.Elements
{
    [UsedImplicitly]
    public class NavigatorElement : Element
    {
        public int Order { get; set; }
        public string Title { get; set; }
        public string MdiIcon { get; set; }
        public string RedirectTo { get; set; }
        public bool MobileView { get; set; }
        public bool GuestOnly { get; set; }
        public bool AuthenticatedOnly { get; set; }
        public RoleType[] PermittedRoles { get; set; }
        public NavigatorElement[] Children { get; set; }
        public Dictionary<string, dynamic> Properties { get; set; }
    }
}

File: projects/Reserve.Client/commons/lib/navigator-element.ts
import { RoleType } from "~/commons/resources/role-type";

export default interface NavigatorElement
{
    title: string;
    order: number;
    mdi_icon?: string;
    redirect_to: string;
    mobile_view: boolean;
    guest_only?: boolean;
    authenticated_only?: boolean;
    permitted_roles?: RoleType[];
    children?: NavigatorElement[];
    properties?: Record<string, any>;
}
